diff -up gtk+-2.24.23/gtk/gtkfilesystem.c.check-cancellable gtk+-2.24.23/gtk/gtkfilesystem.c
--- gtk+-2.24.23/gtk/gtkfilesystem.c.check-cancellable	2014-03-05 23:56:42.000000000 -0500
+++ gtk+-2.24.23/gtk/gtkfilesystem.c	2014-08-22 16:41:49.302999989 -0400
@@ -1445,6 +1445,7 @@ enumerator_files_callback (GObject
			   gpointer      user_data)
 {
   GFileEnumerator *enumerator;
+  GCancellable *cancellable;
   GtkFolderPrivate *priv;
   GtkFolder *folder;
   GError *error = NULL;
@@ -1452,14 +1453,17 @@ enumerator_files_callback (GObject
   GList *file_infos, *f;

   enumerator = G_FILE_ENUMERATOR (source_object);
+
   file_infos = g_file_enumerator_next_files_finish (enumerator, result, &error);
+
+  cancellable = g_object_get_data (G_OBJECT (enumerator), "gtk-folder-cancellable");

-  if (error)
+  if (error || g_cancellable_is_cancelled (cancellable))
     {
-      if (!g_error_matches (error, G_IO_ERROR, G_IO_ERROR_CANCELLED))
+      if (!g_cancellable_is_cancelled (cancellable) && !g_error_matches (error, G_IO_ERROR, G_IO_ERROR_CANCELLED))
         g_warning ("%s", error->message);

-      g_error_free (error);
+      g_clear_error (&error);
       return;
     }

@@ -1522,6 +1526,11 @@ gtk_folder_constructed (GObject *object)
     g_signal_connect (priv->directory_monitor, "changed",
		      G_CALLBACK (directory_monitor_changed), object);

+  g_object_set_data_full (priv->enumerator,
+                          "gtk-folder-cancellable",
+                          g_object_ref (priv->cancellable),
+                          g_object_unref);
+
   g_file_enumerator_next_files_async (priv->enumerator,
				      FILES_PER_QUERY,
				      G_PRIORITY_DEFAULT,
