diff -up gtk+-2.20.1/gtk/gtkfilechooserdefault.c.filechooser-size-column-additional gtk+-2.20.1/gtk/gtkfilechooserdefault.c
--- gtk+-2.20.1/gtk/gtkfilechooserdefault.c.filechooser-size-column-additional	2013-06-06 19:53:33.000000000 -0400
+++ gtk+-2.20.1/gtk/gtkfilechooserdefault.c	2013-06-06 19:55:48.909774327 -0400
@@ -9164,6 +9164,11 @@ search_activate (GtkFileChooserDefault *
   g_assert (impl->search_model == NULL);
 
   search_setup_widgets (impl);
+
+  if (impl->list_size_column)
+    gtk_tree_view_column_set_visible (impl->list_size_column, 
+				      impl->show_size_column);
+
   file_list_set_sort_column_ids (impl);
 }
 
