--- gtk+-2.24.23/gtk/gtkfilechooserdefault.c
+++ gtk+-2.24.23/gtk/gtkfilechooserdefault.c
@@ -2371,9 +2371,27 @@ renderer_editing_canceled_cb (GtkCellRen
 static GtkWidget *
 filter_create (GtkFileChooserDefault *impl)
 {
+  GList *renderer_list;
+  GtkCellRenderer *renderer;
+
   impl->filter_combo = gtk_combo_box_text_new ();
   gtk_combo_box_set_focus_on_click (GTK_COMBO_BOX (impl->filter_combo), FALSE);
 
+  /* we just created it, so the only cell renderer in the combo box
+   * is the text renderer.
+   */
+  renderer_list = gtk_cell_layout_get_cells (GTK_CELL_LAYOUT (impl->filter_combo));
+  if (renderer_list != NULL)
+    {
+      /* set a wrap width of 500px, to avoid applications that set a very long
+       * file filter to overflow the screen
+       */
+      renderer = renderer_list->data;
+      g_object_set (renderer, "wrap-width", 500,
+		    NULL);
+      g_list_free (renderer_list);
+    }
+
   g_signal_connect (impl->filter_combo, "changed",
 		    G_CALLBACK (filter_combo_changed), impl);
 
