--- gtk+-2.24.23/gdk/gdkmarshalers.c
+++ gtk+-2.24.23/gdk/gdkmarshalers.c
@@ -5,7 +5,7 @@
 
 #ifdef G_ENABLE_DEBUG
 #define g_marshal_value_peek_boolean(v)  g_value_get_boolean (v)
-#define g_marshal_value_peek_char(v)     g_value_get_schar (v)
+#define g_marshal_value_peek_char(v)     g_value_get_char (v)
 #define g_marshal_value_peek_uchar(v)    g_value_get_uchar (v)
 #define g_marshal_value_peek_int(v)      g_value_get_int (v)
 #define g_marshal_value_peek_uint(v)     g_value_get_uint (v)
--- gtk+-2.24.23/gtk/gtkmarshal.c
+++ gtk+-2.24.23/gtk/gtkmarshal.c
@@ -6,7 +6,7 @@
 
 #ifdef G_ENABLE_DEBUG
 #define g_marshal_value_peek_boolean(v)  g_value_get_boolean (v)
-#define g_marshal_value_peek_char(v)     g_value_get_schar (v)
+#define g_marshal_value_peek_char(v)     g_value_get_char (v)
 #define g_marshal_value_peek_uchar(v)    g_value_get_uchar (v)
 #define g_marshal_value_peek_int(v)      g_value_get_int (v)
 #define g_marshal_value_peek_uint(v)     g_value_get_uint (v)
--- gtk+-2.24.23/gtk/gtkmarshalers.c
+++ gtk+-2.24.23/gtk/gtkmarshalers.c
@@ -6,7 +6,7 @@
 
 #ifdef G_ENABLE_DEBUG
 #define g_marshal_value_peek_boolean(v)  g_value_get_boolean (v)
-#define g_marshal_value_peek_char(v)     g_value_get_schar (v)
+#define g_marshal_value_peek_char(v)     g_value_get_char (v)
 #define g_marshal_value_peek_uchar(v)    g_value_get_uchar (v)
 #define g_marshal_value_peek_int(v)      g_value_get_int (v)
 #define g_marshal_value_peek_uint(v)     g_value_get_uint (v)
--- gtk+-2.24.23/perf/marshalers.c
+++ gtk+-2.24.23/perf/marshalers.c
@@ -4,7 +4,7 @@
 
 #ifdef G_ENABLE_DEBUG
 #define g_marshal_value_peek_boolean(v)  g_value_get_boolean (v)
-#define g_marshal_value_peek_char(v)     g_value_get_schar (v)
+#define g_marshal_value_peek_char(v)     g_value_get_char (v)
 #define g_marshal_value_peek_uchar(v)    g_value_get_uchar (v)
 #define g_marshal_value_peek_int(v)      g_value_get_int (v)
 #define g_marshal_value_peek_uint(v)     g_value_get_uint (v)
