diff -up gtk+-2.18.9/gtk/gtkfilechooserdefault.c.filechooser-empty-location gtk+-2.18.9/gtk/gtkfilechooserdefault.c
--- gtk+-2.18.9/gtk/gtkfilechooserdefault.c.filechooser-empty-location	2011-02-07 12:23:13.223885166 -0500
+++ gtk+-2.18.9/gtk/gtkfilechooserdefault.c	2011-02-07 13:50:38.748307088 -0500
@@ -7034,6 +7034,9 @@ update_chooser_entry (GtkFileChooserDefa
 
           if (change_entry)
             {
+              g_free(impl->browse_files_last_selected_name);
+              impl->browse_files_last_selected_name = g_strdup (g_file_info_get_display_name (info));
+
               _gtk_file_chooser_entry_set_file_part (GTK_FILE_CHOOSER_ENTRY (impl->location_entry), impl->browse_files_last_selected_name);
 
               if (impl->action == GTK_FILE_CHOOSER_ACTION_SAVE)
@@ -7752,7 +7755,12 @@ gtk_file_chooser_default_get_files (GtkF
       check_save_entry (impl, &info.file_from_entry, &is_well_formed, &is_empty, &is_file_part_empty, &is_folder);
 
       if (is_empty)
-	goto out;
+        {
+          if (!file_list_seen)
+            goto file_list;
+
+          goto out;
+        }
 
       if (!is_well_formed)
 	return NULL;
